### PR TITLE
chore(forms): Support Partial<T>

### LINKED ISSUE #8395: [forms] Should support `Partial<T>` initial values and return validated `T`

### PR DESCRIPTION
**Pull request by ZaymonFC**
- Resolves #8395

### CODE REVIEW

File: packages/ui/react-ui-form/src/hooks/useForm.ts [hunk 1]
```diff
@@ -247,17 +248,24 @@ const flatMap = <T extends BaseObject>(errors: ValidationError[]) => {
   );
 };
 
-// TODO(ZaymonFC): Move to util?
-const getByPath = <T extends object, V>(obj: T, path: FormPath<T>): V => {
+const getByPath = <T extends object, V>(obj: Partial<T>, path: FormPath<T>): V | undefined => {
```
Comment (richburdon) [inline packages/ui/react-ui-form/src/hooks/useForm.ts:251]: i don't understand this notation: prop.3?
if this is for array indirection then use jsonpath notation 'prop[3]' and reuse the jsonpath lib.

File: packages/ui/react-ui-form/src/hooks/useForm.ts [hunk 2]
```diff
@@ -168,8 +169,8 @@ export const useForm = <T extends BaseObject>({
   );
 
   // TODO(burdon): Use path to extract hierarchical value.
-  const getValue = <V>(property: FormPath<T>): V => {
-    return getByPath(values, property);
+  const getValue = <V>(property: FormPath<T>): V | undefined => {
```
Comment (richburdon) [inline packages/ui/react-ui-form/src/hooks/useForm.ts:172]: remove getValue if get/setByPath is used uniformly.

**Review by richburdon — CHANGES_REQUESTED** [review]

File: packages/ui/react-ui-form/src/hooks/useForm.ts (hunk 1 above)
Comment (ZaymonFC) [reply packages/ui/react-ui-form/src/hooks/useForm.ts:251]: I don't understand this comment. Sorry. Can you clarify?

Comment (richburdon) [reply packages/ui/react-ui-form/src/hooks/useForm.ts:251]: how can the second part of the path be a number?

File: packages/common/effect/src/ast.ts [hunk 3]
```diff
@@ -79,7 +79,7 @@ export namespace SimpleType {
 export type JsonProp = string & { __JsonPath: true; __JsonProp: true };
 export type JsonPath = string & { __JsonPath: true };
 
-const PATH_REGEX = /[a-zA-Z_$][\w$]*(?:\.[a-zA-Z_$][\w$]*)*/;
+const PATH_REGEX = /[a-zA-Z_$][\w$]*(?:\.[a-zA-Z_$][\w$]*|\.\d+)*/;
 const PROP_REGEX = /\w+/;
```
Comment (richburdon) [inline packages/common/effect/src/ast.ts:83]: needs a test

File: packages/core/echo/echo-schema/src/types.test.ts [hunk 4]
```diff
@@ -4,11 +4,44 @@
...
+        prop: 'value',
+        arr: [1, 2, 3],
+      },
+    };
+
+    // Basic property access.
+    expect(getValue(obj, 'name' as JsonPath)).toBe('test');
+
+    // Array index access.
+    expect(getValue(obj, 'items.1' as JsonPath)).toBe('b');
```
Comment (richburdon) [inline packages/core/echo/echo-schema/src/types.test.ts:32]: `items[1]` (use jsonpath?)

File: packages/ui/react-ui-form/src/hooks/useForm.ts [hunk 5]
```diff
@@ -168,8 +167,8 @@ export const useForm = <T extends BaseObject>({
   );
 
   // TODO(burdon): Use path to extract hierarchical value.
```
Comment (richburdon) [inline packages/ui/react-ui-form/src/hooks/useForm.ts:166]: remove comment now?

**Review by richburdon — CHANGES_REQUESTED** [review]

File: packages/common/effect/src/ast.ts (hunk 3 above)
Comment (ZaymonFC) [reply packages/common/effect/src/ast.ts:83]: 👍 I'll wait until we settle on the right index syntax then implement.

**Review by richburdon — APPROVED** [review]

**Pull request merged**